pkg/controller/accountclaim/reuse.go

```diff
@@ -36,13 +36,13 @@ func (r *ReconcileAccountClaim) finalizeAccountClaim(reqLogger logr.Logger, acco
 			return err
 		}
 		// Cleanup BYOC secret
-		err = r.removeBYOCSecretFinalizer(accountClaim)
-		if err != nil {
+		secretErr := r.removeBYOCSecretFinalizer(accountClaim)
+		if secretErr != nil {
 			reqLogger.Error(err, "Failed to remove BYOC iamsecret finalizer")
-			return err
+			return secretErr
 		}
 
-		return nil
+		return err
```

Comment:
I think we still want to `return nil` here, as we only want to block if it's not CCS where we can't cleanup.

Reply:
Fair point - I'll update it with a comment then explaining why we're returning nil instead of the err